python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -379,3 +386,214 @@ def get_logs_for_run(
+        backfills.extend(
+            [
+                backfill
+                for backfill in new_backfills
+                if backfill.backfill_timestamp <= created_before
+            ]
+        )
+
+    return backfills[:limit]
+
+
+def get_runs_feed_entries(
+    graphene_info: "ResolveInfo",
+    cursor: Optional[str] = None,
+    limit: Optional[int] = None,
+) -> "GrapheneRunsFeedConnection":
+    """Returns a GrapheneRunsFeedConnection, which contains a merged list of backfills and
+    single runs (runs that are not part of a backfill), the cursor to fetch the next page,
+    and a boolean indicating if there are more results to fetch.
+
+    Args:
+        cursor (Optional[str]): String that can be deserialized into a RunsFeedCursor. If None, indicates
+            that querying should start at the beginning of the table for both runs and backfills.
+        limit (Optional[int]): max number of results to return. If None, will return all results.
+    """
+    from ..schema.backfill import GraphenePartitionBackfill
+    from ..schema.pipelines.pipeline import GrapheneRun
+    from ..schema.runs_feed import GrapheneRunsFeedConnection
+
+    check.opt_str_param(cursor, "cursor")
+    check.opt_int_param(limit, "limit")
+
+    instance = graphene_info.context.instance
+    runs_feed_cursor = RunsFeedCursor.from_string(cursor)
+
+    # if using limit, fetch limit+1 of each type to know if there are more than limit remaining
+    fetch_limit = limit + 1 if limit else None
+    # filter out any backfills/runs that are newer than the cursor timestamp. See RunsFeedCursor docstring
+    # for case when theis is necessary
+    backfills = [
+        GraphenePartitionBackfill(backfill)
+        for backfill in _fetch_backfills_created_before_timestamp(
+            instance,
+            cursor=runs_feed_cursor.backfill_cursor,
+            limit=fetch_limit,
+            created_before=runs_feed_cursor.timestamp,
+        )
+    ]
+    runs = [
+        GrapheneRun(run)
+        for run in _fetch_runs_not_in_backfill(
+            instance,
+            cursor=runs_feed_cursor.run_cursor,
+            limit=fetch_limit,
+            created_before=runs_feed_cursor.timestamp,
+        )
+    ]
+
+    if (
+        fetch_limit and limit
+    ):  # if fetch_limit is non-None then limit must also be non-None, but check for pyright
+        # if we fetched limit+1 of either runs or backfills, we know there must be more results
+        # to fetch on the next call since we will return limit results for this call. Additionally,
+        # if we fetched more than limit of runs and backfill combined, we know there are more results
+        has_more = (
+            len(backfills) == fetch_limit
+            or len(runs) == fetch_limit
+            or len(backfills) + len(runs) > limit
+        )
+    else:
+        # case when limit was not passed, so we fetched all of the results
+        has_more = False
+
+    all_runs = backfills + runs
```

Comment:
nit: all_entries or smth to make it easier to follow that its not just DagsterRuns